rootfs/var/lib/dietpi/services/fs_partition_resize.sh

```diff
@@ -51,7 +51,8 @@
 	# - eMMC:	/dev/mmcblk[0-9]p[1-9]
 	# - NVMe:	/dev/nvme[0-9]n[0-9]p[1-9]
 	# - loop:	/dev/loop[0-9]p[1-9]
-	if [[ $ROOT_DEV == /dev/[shv]d[a-z][1-9] ]]
+	# - Xen/XCP-ng: /dev/xvd[a-z][1-9]
+	if [[ $ROOT_DEV == /dev/[shvx]d[a-z][1-9] ]]
```

Comment:
It does not work like that. It would match xda and vda but not xvda, as the brackets match a single character only. We need to use regex like below:

```suggestion
	if [[ $ROOT_DEV =~ ^/dev/(s|h|v|xv)d[a-z][1-9]$ ]]
```

Reply:
> It does not work like that. It would match xda and vda but not xvda, as the brackets match a single character only. We need to use regex like below:

Great spot on the regex—apologies for missing that nuance! Thanks for the detailed explanation, it makes total sense now. 😅

Not sure how many folks are running DietPi on XCP-ng in their homelabs, but hopefully this helps anyone else in that boat! Let me update the PR with your suggested regex.